lighthouse-core/gather/computed/screenshots.js

```diff
@@ -7,39 +7,26 @@
 
 const ComputedArtifact = require('./computed-artifact');
 
+const SCREENSHOT_TRACE_NAME = 'Screenshot';
+
 class ScreenshotFilmstrip extends ComputedArtifact {
   get name() {
     return 'Screenshots';
   }
 
-  /**
-   * @param {{imageDataPromise: function(): Promise<string>}} frame
-   * @return {Promise<string>}
-   */
-  fetchScreenshot(frame) {
-    return frame
-      .imageDataPromise()
-      .then(data => 'data:image/jpg;base64,' + data);
-  }
-
   /**
    * @param {LH.Trace} trace
-   * @param {LH.ComputedArtifacts} computedArtifacts
    * @return {Promise<Array<{timestamp: number, datauri: string}>>}
   */
-  compute_(trace, computedArtifacts) {
-    return computedArtifacts.requestDevtoolsTimelineModel(trace).then(model => {
-      const filmStripFrames = model.filmStripModel().frames();
-      const frameFetches = filmStripFrames.map(frame => this.fetchScreenshot(frame));
-
-      return Promise.all(frameFetches).then(images => {
-        const result = filmStripFrames.map((frame, i) => ({
-          timestamp: frame.timestamp,
-          datauri: images[i],
-        }));
-        return result;
+  async compute_(trace) {
+    return trace.traceEvents
+      .filter(evt => evt.name === SCREENSHOT_TRACE_NAME)
+      .map(evt => {
+        return {
+          timestamp: evt.ts / 1000,
+          datauri: `data:image/jpg;base64,${evt.args.snapshot}`,
```

Comment:
I'm surprised the compiler doesn't object to accessing `snapshot`

Reply:
well I added snapshot to traceevent definition :)

Reply:
I meant because it was optional but looks like the compiler's ok passing undefined into template strings